c/authService.c

```diff
@@ -173,6 +174,86 @@ static int serveAuthCheck(HttpService *service, HttpResponse *res) {
+    Json *newPassword = jsonObjectGetPropertyValue(inputMessage,"newPassword");
+    int usernameLength = 0;
+    int passwordLength = 0;
+    int newPasswordLength = 0;
+    if(username != NULL) {
+      usernameLength = strlen(jsonAsString(username));
+    }
+    if(password != NULL) {
+      passwordLength = strlen(jsonAsString(password));
+    }
+    if(newPassword != NULL) {
+      newPasswordLength = strlen(jsonAsString(newPassword));
+    }
+    if (usernameLength == 0) {
+      respondWithJsonStatus(response, "No username provided",
+                            HTTP_STATUS_BAD_REQUEST, "Bad Request");
+      return HTTP_SERVICE_FAILED;
+    } else if (passwordLength == 0) {
+      respondWithJsonStatus(response, "No password provided",
+                            HTTP_STATUS_BAD_REQUEST, "Bad Request");
+      return HTTP_SERVICE_FAILED;
+    } else if (newPasswordLength == 0) {
+      respondWithJsonStatus(response, "No new password provided",
+                            HTTP_STATUS_BAD_REQUEST, "Bad Request");
+      return HTTP_SERVICE_FAILED;
+    } else {
+      resetZosUserPassword(jsonAsString(username),  jsonAsString(password), jsonAsString(newPassword), &returnCode, &reasonCode);
+      switch (returnCode) {
+        case PASSWORD_RESET_OK:
+          respondWithJsonStatus(response, "Password Successfully Reset", HTTP_STATUS_OK, "OK");
+          return HTTP_SERVICE_SUCCESS;
+        case PASSWORD_RESET_WRONG_PASSWORD:
+          respondWithJsonStatus(response, "The current password is incorrect. Please try again.",
+                                HTTP_STATUS_UNAUTHORIZED, "Unauthorized");
+          return HTTP_SERVICE_FAILED;
+        case PASSWORD_RESET_WRONG_USER:
+          respondWithJsonStatus(response, "The username entered is not found. Please try again.",
+                                HTTP_STATUS_UNAUTHORIZED, "Unauthorized");
+          return HTTP_SERVICE_FAILED;
+        case PASSWORD_RESET_NO_NEW_PASSSWORD:
+          respondWithJsonStatus(response, "No new password for expired password provided",
+                                HTTP_STATUS_BAD_REQUEST, "Bad Request");
+          return HTTP_SERVICE_FAILED;
+        case PASSWORD_RESET_WRONG_FORMAT:
+          respondWithJsonStatus(response, "The new password format is incorrect. Please try again.",
+                                HTTP_STATUS_BAD_REQUEST, "Bad Request");
+          return HTTP_SERVICE_FAILED;
+        case PASSWORD_RESET_TOO_MANY_ATTEMPTS:
+          respondWithJsonStatus(response,
+                                "An incorrect password has been entered too many times and the account has been locked. Please contact your administrator for further instructions.",
+                                HTTP_STATUS_TOO_MANY_REQUESTS, "Bad Request");
+          return HTTP_SERVICE_FAILED;
+        default:
+          sprintf(responseString, "Password reset FAILED with return code: %d reason code: %d\n", returnCode, reasonCode);
```

Comment:
Can you please use `snprintf` to make sure no new changes will cause a buffer overflow?